interface/codeview.py

```diff
@@ -0,0 +1,26 @@
+import io
+import logging
+from zipfile import ZipFile
+
+from interface.backend.minio_api import MissingFile, download_buffer
+
+log = logging.getLogger(__name__)
+
+
+def submission_cut(request, submission, filename):
+    buff = io.BytesIO()
+    try:
+        download_buffer(f'{submission.pk}.zip', buff)
+    except MissingFile:
+        return io.StringIO("The archive is missing!")
+
+    submission_archive = ZipFile(buff)
+    try:
+        file = submission_archive.read(
+            filename,
+            pwd=None,
+        )
+    except MissingFile:
```

Comment:
I don't think `zipfile.read()` throws this kind of error. Let's leave it as generic as possible and remove `MissingFile` for the moment.